### ISSUE TITLE
no-unbalanced-curlies false positive

### ISSUE THREAD

**Issue by ef4** [ember-template-lint/ember-template-lint#2330]
The no-unbalanced-curlies rule gives a false positive if you have an escaped start curlies and then a newline before the end curlies:

```hbs
\{{foo
}}
```

This diff shows the problem in the test suite:

```diff
diff --git a/test/unit/rules/no-unbalanced-curlies-test.js b/test/unit/rules/no-unbalanced-curlies-test.js
index 056beea1..7b965b4e 100644
--- a/test/unit/rules/no-unbalanced-curlies-test.js
+++ b/test/unit/rules/no-unbalanced-curlies-test.js
@@ -13,6 +13,7 @@ generateRuleTests({
     '\\{{foo}}',
     '\\{{foo}}\\{{foo}}',
     '\\{{foo}}{{foo}}',
+    '\\{{foo\n}}',
   ],
```

I don't really follow why the existing rule is trying to parse one line at a time, so I'm unsure what the best fix is.

**Comment by bmish** [comment]
Sounds like a bug. CC: Original author @pablobm
